openghg/standardise/surface/_agage.py

```diff
@@ -48,46 +47,37 @@ def parse_agage(
     network = clean_string(network)
     instrument = clean_string(instrument)
 
-    # get the parameters from the file metadata, as opposed to from the .json file
-
-    if data is None and filepath is not None:
-        filepath = Path(filepath)
-        with xr.load_dataset(filepath) as dataset:
-            file_params = dataset.attrs
-    elif data is not None:
-        file_params = data.attrs
-
-    # if we're not passed the instrument name, get it from the file:
-
     file_instrument = None
 
-    if "instrument_type" in file_params.keys():
-        # For multiple values in instrument_type of file updating the instrument metadata to multiple
-        instrument_number = len(file_params["instrument_type"].split("/"))
-        if instrument_number > 1:
-            file_instrument = "multiple"
-            instrument = file_instrument
-        else:
-            file_instrument = clean_string(file_params["instrument_type"])
-            if instrument is None:
+    with get_data(dataset=data, filepath=filepath) as dataset:
+        file_attributes = dataset.attrs
+
+        if "instrument_type" in file_attributes.keys():
```

Comment:
You don't need `.keys()` here: `k in some_dict` is the same as `k in some_dict.keys()`.
```suggestion
        if "instrument_type" in file_attributes:
```